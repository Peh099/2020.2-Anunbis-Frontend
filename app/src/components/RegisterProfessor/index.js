import React from 'react';
import { useForm } from 'react-hook-form';
import { yupResolver } from '@hookform/resolvers/yup';
import { useHistory } from 'react-router-dom';
import Form from '../Form';
import Input from '../Input';
import Button from '../Button';
import schema from './validations';
import api from '../../services/Api';

export default function RegisterProfessor({ onRegister }) {
  const history = useHistory();
  const [errorDB, setErrorDB] = React.useState('');
  const { register, handleSubmit, errors } = useForm({
    resolver: yupResolver(schema),
  });

  function onSubmit(data) {
    const body = {
      name: data.name,
      email: data.email,
      password: data.password,
      reg_professor: data.reg_professor,
    };

    api
      .post('/professor', body)
      .then((response) => {
        if (response.status === 201) {
          onRegister('Confirme o seu cadastro com o e-mail de verificação enviado ao seu e-mail.');
        }
      })
      .catch((error) => {
        if (error.response.status === 409) {
          setErrorDB('Professor já cadastrado');
        }
      });
  }

  return (
    <Form onSubmit={handleSubmit(onSubmit)}>
      <Form.Field errorMsg={errors.name?.message}>
        <Input type="text" text="Nome" name="name" register={register} data-testid="input-1-regProf"/>
      </Form.Field>
      <Form.Field errorMsg={errors.reg_professor?.message}>
        <Input
          type="text"
          text="Matrícula"
          name="reg_professor"
          register={register}
          data-testid="input-2-regProf"
        />
      </Form.Field>
      <Form.Field errorMsg={errors.email?.message}>
        <Input
          type="email"
          text="Email Institucional"
          name="email"
          register={register}
          data-testid="input-3-regProf"
        />
      </Form.Field>
      <Form.Field errorMsg={errors.password?.message}>
        {' '}
        <Input
          type="password"
          text="Senha"
          name="password"
          register={register}
          data-testid="input-4-regProf"
        />
      </Form.Field>
      <Form.Field errorMsg={errors.co_password?.message}>
        <Input
          type="password"
          text="Confirmar Senha"
          name="co_password"
          register={register}
          data-testid="input-5-regProf"
        />
      </Form.Field>
      <Form.Field>
        <div className="errorDB">{errorDB}</div>
      </Form.Field>
      <Form.Footer>
        <Button
          text="CANCELAR"
          backColor="#FFF9C4"
          onClick={() => history?.push('/')}
          data-testid="btn-1-regProf"
        />
        <Button
          text="CONFIRMAR"
          type="submit"
          backColor="#FFF9C4"
          onkeydown="Enter"
          data-testid="btn-2-regProf"
        />
      </Form.Footer>
    </Form>
  );
}
