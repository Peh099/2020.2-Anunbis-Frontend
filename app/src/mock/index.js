import MockAdapter from 'axios-mock-adapter';
import api from '../services/Api';


const mock = new MockAdapter(api);

export default mock