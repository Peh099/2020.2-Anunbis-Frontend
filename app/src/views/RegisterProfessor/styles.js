import styled from 'styled-components';


export const Conteiner = styled.div`
  width: 400px;
`;

export const Content = styled.div`
  height: 450px;
  width: 400px;
  display:flex;
  align-items: center;
  flex-direction: column;

  Form {
    height: 300px;
    width: 300px;
    Input{
      margin-bottom: 5px;
      width: 180px;
    }
    Button{
      padding: 12px 8px; 
    }
}
`;

export const Links = styled.div`
  width: 400px;
  display:flex;
  align-items: center;
  justify-content: space-evenly;
  font-size:14px;
  margin-top: 15px;
    
  .btnCadastro {
    margin-right: 70px;
    color: #212121;
  }

  .btnLogin, .btnStudent{
    text-decoration: none;
    color: #212121;
  }
`;

export const Title = styled.h4`
    display: flex;
    justify-content: center;
    align-items: center;
    margin-bottom: 20px;
    margin-top: 40px;
`;