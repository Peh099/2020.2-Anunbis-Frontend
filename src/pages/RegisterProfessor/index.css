.formulario Button{
    margin-top: 120px;
}