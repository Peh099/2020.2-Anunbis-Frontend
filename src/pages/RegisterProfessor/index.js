import React from 'react';
import { useForm } from "react-hook-form";
import { yupResolver } from '@hookform/resolvers/yup';
import './index.css';
import Form from '../../components/Form';
import Input from '../../components/Input';
import Button from '../../components/Button';
import schema from './validations';

export default function RegisterProfessor(){
    const { register, handleSubmit, errors } = useForm({
        resolver: yupResolver(schema),
      });
    
    function onSubmit(data){
        const body = {
            name: data.name,
            email: data.email,
            password: data.password
        }
        console.log(body);
    }

    return (
        <div className="RegisterProfessor">
            <header className="Header">
                <Form title="Cadastro de Professor" onSubmit={handleSubmit(onSubmit)} link="ALUNO?" endereco={`/`}>
                    <Form.Field errorMsg={errors.name?.message}><Input type="text" text="Nome" name="name" register={register} /></Form.Field>
                    <Form.Field errorMsg={errors.email?.message}><Input type="email" text="Email Institucional" name="email" register={register} /></Form.Field>
                    <Form.Field errorMsg={errors.password?.message}> <Input type="password" text="Senha" name="password" register={register} /></Form.Field>
                    <Form.Field errorMsg={errors.co_password?.message}><Input type="password" text="Confirmar Senha" name="co_password" register={register} /></Form.Field>
                    <Form.Footer>
                        <Button text="CANCELAR" />
                        <Button text="CONFIRMAR" type="submit"/>
                    </Form.Footer>
                </Form>                
            </header>
        </div>
    )
}