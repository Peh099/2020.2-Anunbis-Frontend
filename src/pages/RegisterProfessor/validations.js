import * as yup from 'yup';

const schema = yup.object().shape({
    name: yup
    .string()
    .trim()
    .min(2, "Nome deve conter pelo menos dois caracteres")
    .max(254, "Nome deve ter no máximo 254 caracteres.")
    .required("Nome deve ser preenchido."),

    reg_professor: yup
    .string()
    .trim()
    .matches(/^[0-9]*$/, "Matricula deve conter somente números.")
    .required("Matricula deve ser preenchida."),

    email: yup
    .string()
    .trim()
    .lowercase()
    .matches(/^[a-z . 0-9]+@unb.br$/, "Formato do email deve ser: \"matricula(nome)@unb.br\"")
    .max(90, "O email deve ter tamanho máximo de 90 caracteres.")
    .required("Email deve ser preenchido."),

    password: yup
    .string()
    .min(8, "Senha deve ter pelo menos 8 caracteres.")
    .max(100, "Senha deve ter no máximo 100 caracteres.")
    .required("Senha deve ser preenchida."),

    co_password: yup
    .string()
    .oneOf([yup.ref('password')], "Senhas devem ser iguais.")
    .required("Confirmação da senha deve ser preenchida."),
});

export default schema;
