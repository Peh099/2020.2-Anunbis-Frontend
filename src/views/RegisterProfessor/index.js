import React from 'react';
import styled from 'styled-components';
import { useForm } from "react-hook-form";
import { yupResolver } from '@hookform/resolvers/yup';
import Form from '../../components/Form';
import Input from '../../components/Input';
import Button from '../../components/Button';
import schema from './validations';
import { Link, useHistory } from 'react-router-dom';

const Header = ({ children, title }) => {
  const Conteiner = styled.div`
      width: 400px;
  `;

  const Links = styled.div`
      width: 400px;
      display:flex;
      flex-direction: row;
      align-items: center;
      justify-content: space-evenly;
      font-size:14px;

      .btnCadastro {
        margin-right: 70px;
        color: #212121;
      }

      .btnLogin, .btnStudent{
        text-decoration: none;
        color: #212121;
      }
  `;


  const Title = styled.h4`
  display: flex;
  justify-content: center;
  align-items: center;
  margin-bottom: 20px;
  margin-bottom: 10px;
  `;

  return (
    <Conteiner>
      <Links>
        {children}
      </Links>
      <Title>{title}</Title>
    </Conteiner>
  )
}

const Content = styled.div`
height: 450px;
width: 400px;
display:flex;
justify-content: space-evenly;
align-items: center;
flex-direction: column;

Form {
  height: 300px;
  width: 300px;
}
`;

export default function RegisterProfessor() {
  const history = useHistory();
  const [errorDB, setErrorDB] = React.useState('');
  const { register, handleSubmit, errors } = useForm({
    resolver: yupResolver(schema),
  });

  function onSubmit(data) {
    const url = process.env.REACT_APP_API_HOST + "/professor";
    const body = {
      name: data.name,
      email: data.email,
      password: data.password,
      reg_professor: data.reg_professor
    }
    fetch(url, {
      method: 'post',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(body)
    })
      .then(response => response)
      .then(rs => {
        if (rs.ok) {
          setErrorDB("")
          history.push("/login")
        }
        if (rs.status === 409) {
          setErrorDB("Professor já cadastrado")
        }
      })
  };



  return (
    <Content>
      <Header title="Cadastro de Professor">
        <Link className="btnLogin" to="/login">LOGIN</Link>
        <Link className="btnCadastro" to="/">CADASTRO</Link>
        <Link className="btnStudent" to="/">ALUNO?</Link>
      </Header>

      <Form onSubmit={handleSubmit(onSubmit)}>
        <Form.Field errorMsg={errors.name?.message}><Input type="text" text="Nome" name="name" register={register} /></Form.Field>
        <Form.Field errorMsg={errors.reg_professor?.message}><Input type="text" text="Matrícula" name="reg_professor" register={register} /></Form.Field>
        <Form.Field errorMsg={errors.email?.message}><Input type="email" text="Email Institucional" name="email" register={register} /></Form.Field>
        <Form.Field errorMsg={errors.password?.message}> <Input type="password" text="Senha" name="password" register={register} /></Form.Field>
        <Form.Field errorMsg={errors.co_password?.message}><Input type="password" text="Confirmar Senha" name="co_password" register={register} /></Form.Field>
        <Form.Field><div className="errorDB">{errorDB}</div></Form.Field>
        <Form.Footer>
          <Button text="CANCELAR" />
          <Button text="CONFIRMAR" type="submit" />
        </Form.Footer>
      </Form>
    </Content>
  )
}