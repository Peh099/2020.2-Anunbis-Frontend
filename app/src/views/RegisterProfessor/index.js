import React from 'react';
import { useForm } from "react-hook-form";
import { yupResolver } from '@hookform/resolvers/yup';
import Form from '../../components/Form';
import Input from '../../components/Input';
import Button from '../../components/Button';
import schema from './validations';
import { Link, useHistory } from 'react-router-dom';
import api from '../../services/Api';
import {Conteiner, Content, Links, Title} from './styles';

const Header = ({ children, title }) => {
  return (
    <Conteiner>
      <Links>
        {children}
      </Links>
      <Title>{title}</Title>
    </Conteiner>
  )
}

export default function RegisterProfessor() {
  const history = useHistory();
  const [errorDB, setErrorDB] = React.useState('');
  const { register, handleSubmit, errors } = useForm({
    resolver: yupResolver(schema),
  });

  function onSubmit(data) {
    const body = {
      name: data.name,
      email: data.email,
      password: data.password,
      reg_professor: data.reg_professor
    }

    api.post("/professor", body)
      .then(response => {
        if (response.status === 201) {
          history.push("/user/login")
        }
      })
      .catch(error => {
        if (error.response.status === 409) {
          setErrorDB("Professor já cadastrado")
        }
      });
  }


  return (
    <Content>
      <Header title="Cadastro de Professor">
        <Link className="btnLogin" to="/user/login">LOGIN</Link>
        <Link className="btnCadastro" to="/user/student">CADASTRO</Link>
        <Link className="btnStudent" to="/user/student">ALUNO?</Link>
      </Header>

      <Form onSubmit={handleSubmit(onSubmit)}>
        <Form.Field errorMsg={errors.name?.message}><Input type="text" text="Nome" name="name" register={register} /></Form.Field>
        <Form.Field errorMsg={errors.reg_professor?.message}><Input type="text" text="Matrícula" name="reg_professor" register={register} /></Form.Field>
        <Form.Field errorMsg={errors.email?.message}><Input type="email" text="Email Institucional" name="email" register={register} /></Form.Field>
        <Form.Field errorMsg={errors.password?.message}> <Input type="password" text="Senha" name="password" register={register} /></Form.Field>
        <Form.Field errorMsg={errors.co_password?.message}><Input type="password" text="Confirmar Senha" name="co_password" register={register} /></Form.Field>
        <Form.Field><div className="errorDB">{errorDB}</div></Form.Field>
        <Form.Footer>
          <Button text="CANCELAR" backColor="#FFF9C4" />
          <Button text="CONFIRMAR" type="submit" backColor="#FFF9C4" />
        </Form.Footer>
      </Form>
    </Content>
  )
}