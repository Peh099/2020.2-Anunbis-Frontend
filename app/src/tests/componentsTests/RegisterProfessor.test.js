import { screen, render, waitFor, cleanup } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import RegisterProfessor from '../../components/RegisterProfessor';
import mock from '../mock/index';
import '@testing-library/jest-dom/extend-expect';

const mockPost = (number) => {
  if (number === 409) {
    mock.onPost('/professor').reply(409, {
      name: 'professor repetido',
      reg: '12345678911',
      email: '12345678911@unb.br',
      password: '12345678',
    });
  } else {
    mock.onPost('/professor').reply(201, {
      name: 'test',
      reg: '12345678911',
      email: '12345678911@unb.br',
      password: '00000000',
    });
  }
};

const validProf = {
  name: 'test',
  reg: '12345678911',
  email: '12345678911@unb.br',
  password: '00000000',
};

const invalidProf = {
  name: 'professor repetido',
  reg: '12345678911',
  email: '12345678911@unb.br',
  password: '12345678',
};

afterEach(cleanup);

describe('test of UX', () => {
  it('window localtion must be change when click on CANCELAR', () => {
    render(<RegisterProfessor />);
    const buttonCancel = screen.getByText('CANCELAR');
    userEvent.click(buttonCancel);
    expect(window.location.pathname).toEqual('/');
  });

  describe('Tests assyncronous', () => {
    it('Submit Valid Professor', async () => {
      mockPost(201);
      const redirect = jest.fn();
      render(<RegisterProfessor onRegister={redirect} />);

      const inputName = screen.getByPlaceholderText('Nome');
      const inputReg = screen.getByPlaceholderText('Matrícula');
      const inputEmail = screen.getByPlaceholderText('Email Institucional');
      const inputPassword = screen.getByPlaceholderText('Senha');
      const inputCoPassword = screen.getByPlaceholderText('Confirmar Senha');
      const btnConfirm = screen.getByText('CONFIRMAR');

      userEvent.type(inputName, validProf.name);
      userEvent.type(inputReg, validProf.reg);
      userEvent.type(inputEmail, validProf.email);
      userEvent.type(inputPassword, validProf.password);
      userEvent.type(inputCoPassword, validProf.password);
      userEvent.click(btnConfirm);

      await waitFor(() => {
        expect(redirect).toHaveBeenCalled();
      });
    });
  });

  it('Submit Invalid Professor', async () => {
    const redirect = jest.fn();
    render(<RegisterProfessor redirect={redirect} />);
    mockPost(409);

    const inputName = screen.getByPlaceholderText('Nome');
    const inputReg = screen.getByPlaceholderText('Matrícula');
    const inputEmail = screen.getByPlaceholderText('Email Institucional');
    const inputPassword = screen.getByPlaceholderText('Senha');
    const inputCoPassword = screen.getByPlaceholderText('Confirmar Senha');
    const btnConfirm = screen.getByText('CONFIRMAR');

    userEvent.type(inputName, invalidProf.name);
    userEvent.type(inputReg, invalidProf.reg);
    userEvent.type(inputEmail, invalidProf.email);
    userEvent.type(inputPassword, invalidProf.password);
    userEvent.type(inputCoPassword, invalidProf.password);
    userEvent.click(btnConfirm);

    await waitFor(() => {
      expect(screen.getByText('Professor já cadastrado')).toBeInTheDocument();
      expect(redirect).not.toHaveBeenCalled();
    });
  });
});
