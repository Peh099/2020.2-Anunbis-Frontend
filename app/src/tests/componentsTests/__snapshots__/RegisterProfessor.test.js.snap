// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot RegisterProfessor component matches the snapshot 1`] = `
<form
  className="sc-dlnjwi iDwcvG"
  onSubmit={[Function]}
>
  <div
    className="sc-bdnxRM fcTkcg"
  >
    <input
      className="sc-eCApnc doXpOj"
      name="name"
      placeholder="Nome"
      type="text"
    />
    <div
      className="sc-gtsrHT iKVqJa error"
    />
  </div>
  <div
    className="sc-bdnxRM fcTkcg"
  >
    <input
      className="sc-eCApnc doXpOj"
      name="reg_professor"
      placeholder="MatrÃ­cula"
      type="text"
    />
    <div
      className="sc-gtsrHT iKVqJa error"
    />
  </div>
  <div
    className="sc-bdnxRM fcTkcg"
  >
    <input
      className="sc-eCApnc doXpOj"
      name="email"
      placeholder="Email Institucional"
      type="email"
    />
    <div
      className="sc-gtsrHT iKVqJa error"
    />
  </div>
  <div
    className="sc-bdnxRM fcTkcg"
  >
     
    <input
      className="sc-eCApnc doXpOj"
      name="password"
      placeholder="Senha"
      type="password"
    />
    <div
      className="sc-gtsrHT iKVqJa error"
    />
  </div>
  <div
    className="sc-bdnxRM fcTkcg"
  >
    <input
      className="sc-eCApnc doXpOj"
      name="co_password"
      placeholder="Confirmar Senha"
      type="password"
    />
    <div
      className="sc-gtsrHT iKVqJa error"
    />
  </div>
  <div
    className="sc-bdnxRM fcTkcg"
  >
    <div
      className="errorDB"
    >
      
    </div>
    <div
      className="sc-gtsrHT iKVqJa error"
    />
  </div>
  <div
    className="sc-hKFxyN teyFV"
  >
    <button
      className="sc-jSFjdj jthJeV"
      onClick={[Function]}
    >
      CANCELAR
    </button>
    <button
      className="sc-jSFjdj jthJeV"
      type="submit"
    >
      CONFIRMAR
    </button>
  </div>
</form>
`;
